internal/manifest/coreos_installer.go

```diff
@@ -0,0 +1,161 @@
+package manifest
+
+import (
+	"fmt"
+
+	"github.com/osbuild/osbuild-composer/internal/blueprint"
+	"github.com/osbuild/osbuild-composer/internal/osbuild"
+	"github.com/osbuild/osbuild-composer/internal/platform"
+	"github.com/osbuild/osbuild-composer/internal/rpmmd"
+)
+
+type CoreOSInstaller struct {
+	Base
+
+	// Packages to install in addition to the ones required by the
+	// pipeline.
+	ExtraPackages []string
+
+	// Extra repositories to install packages from
+	ExtraRepos []rpmmd.RepoConfig
+
+	platform     platform.Platform
+	repos        []rpmmd.RepoConfig
+	packageSpecs []rpmmd.PackageSpec
+	kernelName   string
+	kernelVer    string
+	product      string
+	version      string
+	Variant      string
+
+	// Biosdevname indicates whether or not biosdevname should be used to
+	// name network devices when booting the installer. This may affect
+	// the naming of network devices on the target system.
+	Biosdevname bool
+
+	FDO *blueprint.FDOCustomization
+}
+
+// NewCOI creates an CoreOS installer pipeline object.
+func NewCOI(m *Manifest,
```

Comment:
Small nit, I think this should be the full name - NewCoreOSInstaller, since this is all in one package that makes it clearer what's being called. 'COI' means I have to figure out which acronym to match against in my head :)

Reply:
That's a good point.  We shouldn't obscure public functions like this.  I'll rename it.